File: packages/sdk/examples/test-sdk.ts
import { Agent } from '../src/index';

async function main() {
    console.log('🧪 Testing SuiLoop TypeScript SDK v0.0.8...');

    const client = new Agent({
        apiKey: process.env.SUILOOP_API_KEY || 'sk_live_your_key_here',
        baseUrl: process.env.AGENT_URL || 'http://localhost:3001'
    });

    try {
        // 1. Health check
        console.log('1. Checking Agent Health...');
        const health = await client.health();
        console.log('✅ Agent Health Status:', health.status);

        // 2. Live market data (Pyth oracle + Scallop APY)
        console.log('\n2. Fetching Live Market Data (Pyth + Scallop)...');
        const market = await client.getMarket();
        console.log(`   SUI Price:       $${market.suiPrice}`);
        console.log(`   Scallop SUI APY: ${market.scallopApy?.supply}%`);
        console.log(`   Navi USDC APY:   ${market.naviUsdcApy?.supply}%`);
        console.log(`   LLM Enabled:     ${market.llmEnabled}`);

        // 3. Execute a SUI flash loan strategy
        console.log('\n3. Executing SUI Flash Loan Strategy...');
        const result = await client.execute('flash-loan-executor', 'SUI');

        if (result.success) {
            console.log('✅ Strategy Executed Successfully!');
            if (result.txHash) console.log(`   TX Hash: ${result.txHash}`);
            if (result.txHash) console.log(`   SuiScan: https://suiscan.xyz/testnet/tx/${result.txHash}`);
            if (result.profit) console.log(`   Yield:   ${result.profit}`);
        } else {
            console.error('❌ Strategy Execution Failed:', result.error);
        }

    } catch (error) {
        console.error('💥 SDK Test Error:', error);
    }
}

main();

File: packages/sdk-python/examples/test_sdk.py
import sys
import os
import json

# Add src to path
sys.path.append(os.path.join(os.path.dirname(__file__), '..', 'src'))

from suiloop.client import Agent

def main():
    print("🧪 Testing SuiLoop Python SDK...")
    
    # Initialize Agent
    agent = Agent(
        api_key="sk_live_your_api_key_here",
        base_url="http://localhost:3001"
    )

    try:
        # 1. Check Health
        print("1. Checking Agent Health...")
        health = agent.health()
        print(f"✅ Agent Health Status: {health.get('status')}")

        # 2. Get Market Info
        print("\n2. Fetching Market State...")
        market = agent.get_market()
        print(f"✅ SUI Price: ${market.get('suiPrice')}")
        print(f"✅ Gas Price: {market.get('gasPrice')} MIST")

        # 3. Execute Strategy
        print("\n3. Executing Flash Loan Strategy...")
        result = agent.execute(
            strategy="flash-loan-executor",
            params={"amount": 0.1},
            asset="SUI"
        )

        if result.get("success"):
            print("✅ Strategy Executed Successfully!")
            print(f"🔗 Hash: {result.get('txHash')}")
        else:
            print(f"❌ Strategy Execution Failed: {result.get('error')}")

    except Exception as e:
        print(f"💥 SDK Test Error: {e}")

if __name__ == "__main__":
    main()
